php/classes/DB.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+require_once __DIR__ . '/../inc/configDB.php';
+
+class DB {
+  // (A) CONSTRUCTOR - CONNECT TO DATABASE
+  private $pdo = null;
+  private $stmt = null;
+  public $msg = "";
+  function __construct () {
+    try {
+      $this->pdo = new PDO(
+        "mysql:host=".DB_HOST.";dbname=".DB_NAME.";charset=".DB_CHARSET, 
+        DB_USER, DB_PASSWORD, [
+          PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION,
+          PDO::ATTR_DEFAULT_FETCH_MODE => PDO::FETCH_ASSOC
+        ]
+      );
+    } catch (Exception $ex) { die($ex->getMessage()); }
+  }
+
+  // (B) DESTRUCTOR - CLOSE DATABASE CONNECTION
+  function __destruct () {
+    if ($this->stmt!==null) { $this->stmt = null; }
+    if ($this->pdo!==null) { $this->pdo = null; }
+  }
+
+  function queryFetchAll ($sql) {
+    return $this->pdo->query($sql)->fetchAll(PDO::FETCH_ASSOC);
+  }
+
+  function queryFetch ($sql) {
+    return $this->pdo->query($sql)->fetch();
+  }
+
+  function getFirstReceiptToCheck() {
+    $sql = "SELECT * 
+    FROM receipts
+    WHERE checked=false
+    ORDER BY `date_emission` ASC, `id` ASC
+    LIMIT 1;";
+      
+    return $this->pdo->query($sql)->fetch(PDO::FETCH_ASSOC);
+  }
+
+  function saveReceipt() {
+    try {
+        $_POST['ischecked'] === "true" ? $isChecked = 1 : $isChecked = 0;
```

Comment:
I don't think your `DB` class should be the one checking your requests variables. Also, your function is doing too many things. There is a lot of possible implementations that are better respecting the [SOLID principles](https://en.wikipedia.org/wiki/SOLID), but I think the following proposition is fairly simple and des respect these principles better

```
function createReceipt($id, $date, $isChecked...) {
  // Only create a query and save your receipt ni database here
  // Every other bit of logic either goes to the controller or to another class
}
function updateReceipt($id, $date, $isChecked...) {
  // Only create a query and save your receipt ni database here
  // Every other bit of logic either goes to the controller or to another class
}
```